Add indexing optimization guide (#2720)
meilisearch/documentation: learn/cookbooks/indexing_best_practices.mdx

---
title: Indexing best practices — Meilisearch documentation
description: All the tips to speed up your documents indexing process.
---

# Indexing best practices

> In this guide, we will review the best practices to help you index your data efficiently and speed up the indexing process.

Reviewer: ```suggestion
In this guide, you will find some the best practices to index your data efficiently and speed up the indexing process.
```

To speed up indexing, review your list of [searchable attributes](/learn/configuration/displayed_searchable_attributes#searchable-fields) and ensure it includes only the fields you want to be checked for query word matches. This improves both relevance and search speed by removing irrelevant data from your database. It will also keep your disk usage to the necessary minimum.

### Review filterable and sortable attributes

Some document fields are necessary for [filtering](/learn/fine_tuning_results/filtering) and [sorting](/learn/fine_tuning_results/sorting) results, but they dot not need to be _searchable_. Generally, **numeric and boolean fields** fall into this category. Make sure to review your list of searchable attributes and remove any fields that are only used for filtering or sorting.

## Set ranking rules before indexing

Whenever you update your [ranking rules](/learn/core_concepts/relevancy), Meilisearch will trigger a reindexing of all your documents. This can be a time-consuming process, especially if you have a large dataset. For this reason, it is better to define your ranking rules before indexing your data.

## Optimize document size

> Smaller documents are processed faster—so make sure to trim down any unncessary data from your documents. When a document field is missing from the list of [searchable](/reference/api/settings#searchable-attributes), [filterable](/reference/api/settings#filterable-attributes), [sortable](/reference/api/settings#sortable-attributes), or [displayed](/reference/api/settings#displayed-attributes) attributes, it might be best to remove it from the document. To go further, consider compressing your data using methods such as `br`, `deflate`, or `gzip` supported by Meilisearch (see the [supported encoding formats](/reference/api/overview#content-encoding).)

Reviewer: ```suggestion
Smaller documents are processed faster, so make sure to trim down any unncessary data from your documents. When a document field is missing from the list of [searchable](/reference/api/settings#searchable-attributes), [filterable](/reference/api/settings#filterable-attributes), [sortable](/reference/api/settings#sortable-attributes), or [displayed](/reference/api/settings#displayed-attributes) attributes, it might be best to remove it from the document. To go further, consider compressing your data using methods such as `br`, `deflate`, or `gzip`. Consult the [supported encoding formats reference](/reference/api/overview#content-encoding).
```

## Keep your Meilisearch up-to-date!

> The team at Meilisearch is constantly working to improve the indexing process. Make sure to keep your Meilisearch instance up-to-date to benefit from the latest improvements, like the [2x speed increase in v1.6](https://blog.meilisearch.com/meilisearch-1-6/#performance-optimization). You can see a list of all our releases [here](https://blog.meilisearch.com/tag/release/).

Reviewer: I really don't think users will gain much from going to the release blog post of a specific Meilisearch version. Also, this information will grow stale as time goes on, so better not to have to worry about maintaining it.

I'm also not sure about linking to the blog for a list of releases. From a developer POV, I believe the GitHub releases page is more concise (it also includes patch releases, which sometimes have performance improvements).

Oh, final thing: avoid using "here" as link text. In terms of usability and readability, it's always better to have the link describing the page users will reach if they click on it. I also read it might have a positive impact on SEO, since "the releases page on github" has a lot more semantically meaningful content than "here".

```suggestion
The team at Meilisearch is constantly working to improve the indexing process. Make sure to keep your Meilisearch instance up-to-date to benefit from the latest improvements. You can see [a list of all our engine releases on GitHub](https://github.com/meilisearch/meilisearch/releases?q=prerelease%3Afalse).
```

> The team at Meilisearch is constantly working to improve the indexing process. Make sure to keep your Meilisearch instance up-to-date to benefit from the latest improvements, like the [2x speed increase in v1.6](https://blog.meilisearch.com/meilisearch-1-6/#performance-optimization). You can see a list of all our releases [here](https://blog.meilisearch.com/tag/release/).

Reviewer: ```suggestion
Make sure to keep your Meilisearch instance up-to-date to benefit from the latest improvements, like the [2x speed increase in v1.6](https://blog.meilisearch.com/meilisearch-1-6/#performance-optimization). You can see a list of all our releases [here](https://blog.meilisearch.com/tag/release/).
```

<Capsule intent="note">
For a comprehensive explanation of how indexing works under the hood, read our [indexing deep dive](https://blog.meilisearch.com/best-practices-for-faster-indexing/) article.
</Capsule>

> By following these recommendations, you can optimize the indexing process of Meilisearch and enhance the overall search experience for your users. Go build something great! 🚀

Reviewer: Here I am, sucking the joy of every piece of content every written in the history of mankind.

```suggestion
By following these recommendations, you can optimize the indexing process of Meilisearch and enhance the overall search experience for your users.
```

---
title: Indexing best practices — Meilisearch documentation
description: Tips to speed up your documents indexing process.
sidebarDepth: 3
---

# Indexing best practices

> In this guide, you will find some the best practices to index your data efficiently and speed up the indexing process.

Reviewer 2: ```suggestion
In this guide, you will find some of the best practices to index your data efficiently and speed up the indexing process.
```
